server/src/utils.ts

```diff
@@ -273,10 +297,11 @@ export let parseCompilerLogOutput = (
 
 	let result: filesDiagnostics = {};
 	parsedDiagnostics.forEach((parsedDiagnostic) => {
-		let [fileAndLocation, ...diagnosticMessage] = parsedDiagnostic.content;
-		let locationSeparator = fileAndLocation.indexOf(":");
-		let file = fileAndLocation.substring(2, locationSeparator);
-		let location = fileAndLocation.substring(locationSeparator + 1);
+		let [fileAndLocationLine, ...diagnosticMessage] = parsedDiagnostic.content;
+
+		let fileAndLocation = fileAndLocationLine.trim()
```

Comment:
Why the trim here?

Reply:
I saw that the lines came in with spacing in front of the path, like: '  C:\\rescript-project-template\\src\\Demo.res:1:14-20\r\r'. Would have to remove the whitespace so it's possible to slice off the drive letter and colon afterwards. I think this was done with .slice(2) before. Would you rather it be done using that method? Is it always 2 spaces?

Reply:
It was `substring(2)` but that's the issue we saw. It can be done in the original way.

I do feel like trim makes it clearer what's going on, and can be more robust if the compiler decides to change the number of spaces (like @mewhhaha seems to be implying from 'Is it always 2 spaces')

Let us know which you prefer and it's an easy fix.

Reply:
Ah yes, the 2 spaces is guaranteed. I've put a note here just now: https://github.com/rescript-lang/rescript-vscode/blob/20841214830a3a81140f63a8973ab5d08cfcd79f/CONTRIBUTING.md#compilerlog

Whenever it's not, it's a bug that shouldn't be covered up here, but rather fixed int the compiler. We've caught a few compiler diagnostics display bugs this way actually.

In general also, we'd like to avoid trimming: https://github.com/rescript-lang/rescript-vscode/blob/20841214830a3a81140f63a8973ab5d08cfcd79f/CONTRIBUTING.md#formatting-newline

FYI trimming has caused and hidden all sorts of bugs before. Sometime the formatting adds a newline, or two, then the editor trims, adds a new line, or whatever. Lots of little shuffling.

Reply:
I'll go ahead and fix this and change it back to the cutting out the 2 spaces using substring.